luarules/gadgets/unit_collision_damage_behavior.lua

```diff
@@ -161,23 +186,34 @@ function gadget:UnitUnloaded(unitID, unitDefID, unitTeam,  transportID, transpor
 	fallingUnits[unitID] = true --units falling from transports should take collision damagee from any trajectory, including when bouncing off of other objects.
 end
 
+function gadget:UnitEnteredAir(unitID, unitDefID, unitTeam)
+	if not transportedUnits[unitID] and not unitDefData[unitDefID].canFly then
+		launchedUnits[unitID] = true
+	end
+end
+
 function gadget:UnitLeftAir(unitID, unitDefID, unitTeam)
 	fallingUnits[unitID] = nil
+	launchedUnits[unitID] = nil
 end
 
 function gadget:UnitDestroyed(unitID, unitDefID, unitTeam, attackerID, attackerDefID, attackerTeam)
 	transportedUnits[unitID] = nil
 	unitInertiaCheckFlags[unitID] = nil
+	launchedUnits[unitID] = nil
+	fallingUnits[unitID] = nil
 end
 
 function gadget:GameFrame(frame)
-	for unitID, expirationFrame in pairs(unitInertiaCheckFlags) do
+	for unitID, data in pairs(unitInertiaCheckFlags) do
 		if not transportedUnits[unitID] and not spGetUnitIsDead(unitID) then
 			local velX, velY, velZ, velocityLength = spGetUnitVelocity(unitID)
-			
-			if velocityLength > velocityCap then -- Sheer off extreme velocities within acceptable range
+			if not data.velocityReduced and velocityLength > data.velocityCap then -- Sheer off extreme velocities within acceptable range
```

Comment:
This comment can be deleted
```suggestion
			if not data.velocityReduced and velocityLength > data.velocityCap then
```